packages/@o3r/extractors/src/core/builder.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  BuilderContext,
+  BuilderOutput,
+  createBuilder as createNgBuilder
+} from '@angular-devkit/architect';
+import { getEnvironmentInfo } from '@o3r/schematics';
+import { performance } from 'node:perf_hooks';
+
+/**
+ * Wrapper method of `createBuilder` from `@angular-devkit-architect`
+ * to handle metrics around the builder run
+ * @param builderFn
+ */
+export const createBuilder = <S, O extends BuilderOutput = BuilderOutput>(
```

Comment:
Maybe having UTs for this function (and the Schematic similar one) can be valuable